core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3064,28 +3080,62 @@ object Stream extends StreamLowPriority {
     *
-    * Alias for `sleep(d).repeat`.
+    * Alias for `sleep(period).repeat`.
     */
-  def fixedDelay[F[_]](d: FiniteDuration)(implicit t: Temporal[F]): Stream[F, Unit] =
-    sleep(d).repeat
+  def fixedDelay[F[_]](period: FiniteDuration)(implicit t: Temporal[F]): Stream[F, Unit] =
+    sleep(period).repeat
 
-  /** Discrete stream that emits a unit every `d`.
+  /** Discrete stream that emits a unit every `d`, with missed period ticks dampened.
     *
     * See [[fixedDelay]] for an alternative that sleeps `d` between elements.
     *
-    * @param d FiniteDuration between emits of the resulting stream
+    * @param period duration between emits of the resulting stream
     */
-  def fixedRate[F[_]](d: FiniteDuration)(implicit t: Temporal[F]): Stream[F, Unit] = {
-    def now: Stream[F, Long] = Stream.eval(t.monotonic.map(_.toNanos))
-    def go(started: Long): Stream[F, Unit] =
-      now.flatMap { finished =>
-        val elapsed = finished - started
-        Stream.sleep_(d - elapsed.nanos) ++ now.flatMap { started =>
-          Stream.emit(()) ++ go(started)
-        }
+  def fixedRate[F[_]](period: FiniteDuration)(implicit t: Temporal[F]): Stream[F, Unit] =
+    fixedRate(period, true)
+
+  /** Discrete stream that emits a unit every `d`.
+    *
+    * See [[fixedDelay]] for an alternative that sleeps `d` between elements.
+    *
+    * This operation differs in that the time between ticks should roughly be equal to the specified period, regardless
+    * of how much time it takes to process that tick downstream. For example, with a 1 second period and a task that takes 100ms,
+    * the task would run at timestamps, 1s, 2s, 3s, etc. when using `fixedRate >> task` whereas it would run at timestamps
+    * 1s, 2.1s, 3.2s, etc. when using `fixedDelay >> task`.
+    *
+    * In the case where task processing takes longer than a single period, 1 or more ticks are immediately emitted to "catch-up".
+    * The `dampen` parameter controls whether a single tick is emitted or whether one per missed period is emitted.
+    *
+    * @param period period between emits of the resulting stream
+    * @param dampen true if a single unit should be emitted when multiple periods have passed since last execution, false if a unit for each period should be emitted
+    */
+  def fixedRate[F[_]](period: FiniteDuration, dampen: Boolean)(implicit
+      F: Temporal[F]
+  ): Stream[F, Unit] =
+    Stream.eval(F.monotonic.map(_.toMillis)).flatMap(t => fixedRate_(period.toMillis, dampen, t))
+
+  private def getMonotonicMillis[F[_]](implicit F: Temporal[F]): Stream[F, Long] =
+    Stream.eval(F.monotonic.map(_.toMillis))
+
+  private def fixedRate_[F[_]: Temporal](
+      periodMillis: Long,
+      dampen: Boolean,
+      t: Long
+  ): Stream[F, Unit] =
+    getMonotonicMillis.flatMap { now =>
+      val next = t + periodMillis
+      if (next <= now) {
+        val cnt = (now - next + periodMillis - 1) / periodMillis
```

Comment:
Since `next = t + periodMillis` this should be the same as 
```suggestion
        val cnt = (now - t - 1) / periodMillis
```